Assets/_Scripts/TurnController.cs

```diff
@@ -294,12 +293,79 @@ void Update()
         return true;
     }
 
-    void OnDestroy() {
+    private void onWaveComplete() {
+        StageInfoContainer stageInfo = GameManager.Instance.gameState.GetCurrentStage();
+        WaveInfoContainer waveInfo = stageInfo.GetWaveInfo(this.field.currentWaveIndex);
+
+        this.field.currentWaveIndex++;
+        if (this.field.currentWaveIndex >= stageInfo.numWaves) {
+            GameManager.Instance.gameState.SetNextStageIndex(GameManager.Instance.gameState.progressData.currentStageIndex + 1); // May want to change logic in the future
```

Comment:
Not exactly a good design choice here. TurnController is only responsible for a single stage - and how the game is played out. It should not know that progression even exists in the game. 

Perhaps the TurnController can look into taking in a callback which can progress the game.

Reply:
I can make some sort of StageProgressor class, that will only contain this logic, and will be called here in TurnController anyway.

Reply:
It's called StageController now